### DISCUSSION TITLE
Support for ARM9

### DISCUSSION THREAD

**Discussion by leo60228** [oxidecomputer/hubris#370] [Ideas]
> This means that, at the moment, Hubris won't run on your Game Boy Advance, Nintendo DS, or Apple Newton.

The DS has an MPU, so it's further along than the docs make it sound... I doubt anyone has a burning desire to actually use hubris on a DS, though.

**Comment by cbiffle** [comment]
The DS likely has an _MMU,_ so we'd need to generate pagetables etc. to use it like an M-series MPU.

**Comment by mx-shift** [comment]
AFAIK Nintendo DS has an ARM946E-S and an ARM7TDMI both w/o MMU.

**Reply by cbiffle** [reply]
That was my understanding as well, but I've never developed for the DS. If that's the case, Hubris is definitely not the right platform.

**Reply by leo60228** [reply]
The ARM7TDMI is identical to the GBA's which has neither an MMU or MPU, but I believe the DS's ARM946E-S has an MPU with no MMU (see [gbatek](https://mgba-emu.github.io/gbatek/#armcp15protectionunitpu)).

**Reply by cbiffle** [reply]
Funky! I've never seen one of those on an ARM9 core before. That looks very similar to the MPU later standardized in the M/R architecture profiles, though only at a conceptual level -- it's accessed rather differently and has some register layout differences, if I'm reading this correctly.

You might be able to get that working. To manage expectations, we are unlikely to accept such changes upstream -- we don't use ARM9 cores and I'm pretty sure my DS needs a new battery. :-)
